x-pack/plugins/security_solution/server/lib/detection_engine/signals/search_after_bulk_create.ts

```diff
@@ -234,33 +229,18 @@ export const searchAfterAndBulkCreate = async ({
           logger.debug(
             buildRuleMessage(`filteredEvents.hits.hits: ${filteredEvents.hits.hits.length}`)
           );
+        }
 
-          if (
-            filteredEvents.hits.hits[0].sort != null &&
-            filteredEvents.hits.hits[0].sort.length !== 0
-          ) {
-            sortId = filteredEvents.hits.hits[0].sort
-              ? filteredEvents.hits.hits[0].sort[0]
-              : undefined;
-          } else {
-            logger.debug(buildRuleMessage('sortIds was empty on filteredEvents'));
-            toReturn.success = true;
-            break;
-          }
+        // we are guaranteed to have searchResult hits at this point
+        // because we check before if the totalHits or
+        // searchResult.hits.hits.length is 0
+        const lastSortId = searchResult.hits.hits[searchResult.hits.hits.length - 1].sort;
+        if (lastSortId != null && lastSortId.length !== 0) {
+          sortId = lastSortId[0];
```

Comment:
In the next PR we probably want to carry the whole `.sort` array instead of just element 0 through to the next query so that we keep the sort tiebreakers as well.

Reply:
Yeah I agree with that as well. It seems "off" or I might be missing something that the code is currently only pulling the first sort element from it. I can do that in this PR (no big rush) if it makes sense. Will get feedback from @dhurley14